src/render/terrain.rs

```diff
@@ -1067,6 +1097,20 @@ impl Context {
                     }
                 }
                 staging_buf.unmap();
+                
+                {
+                    let mut mapping = self.terrain_data.slice(..).get_mapped_range_mut();
```

Comment:
we shouldn't be mapping terrain_data. In fact, it's not mappable. It only has `mapped_at_creation` for the initial data. The rest of the operations should be on GPU.